dropwizard-jetty/src/main/java/io/dropwizard/jetty/ZipExceptionHandlingInputStream.java

```diff
@@ -0,0 +1,97 @@
+package io.dropwizard.jetty;
+
+import java.io.EOFException;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.zip.ZipException;
+
+import org.eclipse.jetty.http.BadMessageException;
+
+/**
+ * This InputStream is used to decorate a GZIPInputStream or InflaterInputStream, intercept decompression
+ * exceptions, and rethrow them as BadMessageExceptions
+ */
+class ZipExceptionHandlingInputStream extends FilterInputStream {
```

Comment:
I would prefer to extend this class from `InputStream` and declare the `InputStrem` as a field instead of extending from `FilterInputStream`. Using `FilterInputStream` doesn't seem to provide any advantage considering we override all the methods anyway to handle exceptions.

Reply:
It did cover `mark()` and `markSupported()`, but reimplementing them was pretty easy.